woocommerce/woocommerce-ios
# Coupons: add coupon discard alert

Summary
==========
Fix issue #6836 by defining a discard dialog before the merchant dismisses the Edit coupon view from a swipe-down gesture. This is made possible through wrapping the `AddEditCoupon` SwiftUI view in a `UIHostingController`, where we can implement the `UIAdaptivePresentationControllerDelegate` protocol and register the hosting controller as a presentation controller, making it possible to handle the gesture from in an UIKit manner, since SwiftUI doesn't offer us a reliable way to do so.

Screenshots
==========

https://user-images.githubusercontent.com/5920403/169617487-ed581954-c418-4a2e-bf2c-03c704d19bfe.mov



How to Test
==========
1. Make sure the Coupons experimental toggle is active
2. Go to the Coupons section and in the Coupon list, select an editable one
3. Once inside the select Coupon details, click the action menu button and select `Edit Coupon`
4. Make any changes to the coupon and try to do a swipe-down to dismiss in the edit view
5. Verify if the discard warning is displayed
6. Remove all changes done and try the swipe-down again, verify that now the dismiss happens without issues

Update release notes:
